marmelab/react-admin
# [ui-materialui]: Hide `<AutoCompleteInput>` "Create" option for empty input

## Problem

The “create option on-the-fly” UI of Autocomplete inputs is confusing. I can create an option with "" as value.
Fixes #10043

## Solution

- [x] Don’t display “Create” on the input unless the user types something
- [x] Fix the brief flash of `Create xxx` -> `xxx`

## How To Test

- run `ONLY=ra-ui-materialui make storybook`
- go to `http://localhost:9010/?path=/story/ra-ui-materialui-input-autocompleteinput--on-create`
- clear the input and see that you cannot create an empty option

## Additional Checks

~- [ ] The PR includes one or several **stories** (if not possible, describe why)~
~- [ ] The **documentation** is up to date~

## Screencasts

Before flash fix:
[Capture vidéo du 2024-09-25 19-42-25.webm](https://github.com/user-attachments/assets/824e9575-9473-4dff-8663-f06d8d4d6f1d)
After:
[Capture vidéo du 2024-09-25 19-43-16.webm](https://github.com/user-attachments/assets/12f1c759-60f6-4f12-90a9-2d7aa8fcaf7e)
